### ISSUE TITLE
The binary operator Equal is not defined for the types 'System.Nullable`1[System.Guid]' and 'System.Guid'.

### ISSUE THREAD

**Issue by ChangedHourly** [aspnet/EntityFramework#3782]
Hi,
I have come across a problem running a query to get the count(s) of child collections, the exact same code is OK with EF6:

                IQueryable<IRClaim> dbQuery = _context.Set<IRClaim>().AsNoTracking();

                var counts = (from c in dbQuery
                              select new
                              {
                                  CountA = c.Collection1.Count,
                                  CountB = c.Collection2.Count,
                                  CountC = c.Collection3.Count
                              }).FirstOrDefault();

The error is:
"The binary operator Equal is not defined for the types 'System.Nullable`1[System.Guid]' and 'System.Guid'."

I have just updated the model in my EF7 solution to use the Fluent API for configuration and I haven't reflected these changes in my EF6 solution yet, but I didn't think this would be related(?). Despite the different configuration (Annotations vs Fluent API) the models produce the same database structure.

**Comment by maumar** [comment]
Can you include code for your DbContext and the entities used in the query?

**Comment by ChangedHourly** [comment]
Hi, I have some issues with sharing this publicly, but I can share it privately if that is OK.

**Comment by maumar** [comment]
@ChangedHourly  I sent you an email to which you can reply with the repro code

**Comment by maumar** [comment]
This issue has been fixed in the current bits, probably by the following commit:
https://github.com/aspnet/EntityFramework/commit/41ad57ad2f3755c5e35b51d939fab52b489ce738

However, the repro uncovers a different issue, most likely a duplicate of #3616 
Will keep it open, just in case, until we are sure that those two issues have the same root cause.

**Comment by maumar** [comment]
repro for the second issue:

classes:

```
    public class Foo
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public ICollection<Bar> Bars { get; set; }
        public ICollection<Baz> Bazs { get; set; }
    }

    public class Bar
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    public class Baz
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }
```

query:

```
ctx.Foos.Select(f => new { c1 = f.Bars.Count, c2 = f.Bazs.Count }).ToList();
```

sql we produce is the following:

```
SELECT (
    SELECT COUNT(*)
    FROM [Baz] AS [b]
    WHERE [f].[Id] = [b].[FooId]
)
FROM [Foo] AS [f]
```

**Comment by maumar** [comment]
Fixed in e697bd49359b50e4a385f199c0e4aabff769aca2
